packages/manager/apps/container/src/payment-modal/PaymentModal.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react';
+
+import { Modal } from 'react-bootstrap';
+import { useTranslation } from 'react-i18next';
+
+import { useShell } from '@/context';
+
+import { fetchIcebergV6 } from '@ovh-ux/manager-core-api';
+
+import './styles.scss';
+
+interface IPaymentMethod {
+  icon?: any;
+  label: string;
+  status: string;
+  default: boolean;
+  oneclick: boolean;
+  lastUpdate: string;
+  description: string;
+  integration: string;
+  paymentType: string;
+  creationDate: string;
+  paymentMeanId: number;
+  expirationDate?: string;
+  paymentSubType: string;
+  paymentMethodId: number;
+}
+
+interface IPaymentMethodResponse {
+  data: IPaymentMethod[];
+}
+
+const PaymentModal = (): JSX.Element => {
+  const NO_DEFAULT_CARD = 'NO_DEFAULT_CARD';
+  const EXPIRED_CARD = 'EXPIRED_CARD';
+  const [mode, setMode] = useState('');
+  const { t } = useTranslation('payment-modal');
+  const [showPaymentModal, setShowPaymentModal] = useState(false);
+  const shell = useShell();
+
+  const paymentMethodURL = shell
+    .getPlugin('navigation')
+    .getURL('dedicated', '#/billing/payment/method');
+
+  useEffect(() => {
+    const getPaymentMethods = async () => {
+      const paymentResponse:IPaymentMethodResponse = await fetchIcebergV6({route: '/me/payment/method'});
+      const hasDefaultPaymentMethod = paymentResponse.data && paymentResponse.data.length > 0 && paymentResponse.data.find(currentPaymentMethod => currentPaymentMethod.default);
+      const hasExpiredCreditCard = paymentResponse.data && paymentResponse.data.length > 0 && paymentResponse.data.find(currentPaymentMethod => currentPaymentMethod.paymentType === 'CREDIT_CARD' && currentPaymentMethod.expirationDate && new Date(currentPaymentMethod.expirationDate).getTime() < Date.now());
+      if (!hasDefaultPaymentMethod || hasExpiredCreditCard) {
+        setMode(hasExpiredCreditCard ? EXPIRED_CARD : NO_DEFAULT_CARD);
+        setShowPaymentModal(true);
+      }
+    }
+    getPaymentMethods();
```

Comment:
We should be able to just execute the code inside getPaymentMethods

Reply:
You can't use an async function directly inside a useEffect hook.